ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PullProjectNode.java

```diff
@@ -65,7 +67,7 @@
   private final ImmutableList<SelectExpression> selectExpressions;
   private final LogicalSchema outputSchema;
   private final LogicalSchema intermediateSchema;
-  private final List<ExpressionMetadata> compiledSelectExpressions;
+  private final List<ExpressionEvaluator> compiledSelectExpressions;
```

Comment:
nit: It's a bit odd to let a `metadata` implements an `evaluator` interface; how about just rename `ExpressionMetadata` to `CompiledExpression` or `EvaluableExpression`?